nordicsemi/dfu/dfu_trigger.py

```
@@ -0,0 +1,128 @@
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+# Importing libusb1 library
+import os
+working_dir = os.getcwd()
+file_dir = os.path.dirname(__file__)
+os.chdir(file_dir)
+os.chdir("../../libusb")
+import usb1
+os.chdir(working_dir)
+
+
+from pc_ble_driver_py.exceptions    import NordicSemiException
+
+LIBUSB_ENDPOINT_IN = 0x80
+LIBUSB_ENDPOINT_OUT = 0x00
+LIBUSB_REQUEST_TYPE_STANDARD = 0x00 << 5
+LIBUSB_REQUEST_TYPE_CLASS = 0x01 << 5
+LIBUSB_REQUEST_TYPE_VENDOR = 0x02 << 5
+LIBUSB_REQUEST_TYPE_RESERVED = 0x03 << 5
+LIBUSB_RECIPIENT_DEVICE = 0x00
+LIBUSB_RECIPIENT_INTERFACE = 0x01
+LIBUSB_RECIPIENT_ENDPOINT = 0x02
+LIBUSB_RECIPIENT_OTHER = 0x03
+
+ReqTypeInterfaceClass = LIBUSB_REQUEST_TYPE_CLASS | LIBUSB_RECIPIENT_INTERFACE;
+ReqTypeIN = ReqTypeInterfaceClass | LIBUSB_ENDPOINT_IN;
+ReqTypeOUT = ReqTypeInterfaceClass | LIBUSB_ENDPOINT_OUT;
+NORDIC_SEM_VER_REQUEST = 8;
+NORDIC_DFU_INFO_REQUEST = 7;
+DFU_DETACH_REQUEST = 0;
+
+class DFUTrigger:
+    def __init__(self):
+        self.context = usb1.USBContext()
+
+    def clean(self):
+        self.context.close()
+
+    def select_device(self, listed_device):
+        all_devices = self.context.getDeviceList()
+        filtered_devices = [dev for dev in all_devices\
+        if hex(dev.getVendorID())[2:].lower() == listed_device.vendor_id.lower() and \
+        hex(dev.getProductID())[2:].lower() == listed_device.product_id.lower()]
+
+        access_error = False
+
+        for nordic_device in filtered_devices:
+            try:
+                handle = nordic_device.open()
+                SNO = handle.getSerialNumber()
+                handle.close()
+                if (SNO.lower() == listed_device.serial_number.lower()):
+                    return nordic_device
+
+            except usb1.USBErrorNotFound as err:
```

Could add comment to explain why this error is swallowed.